samples/tutorials/mnist/02_Local_Development_with_Docker_Image_Components.ipynb

```diff
@@ -0,0 +1,514 @@
+    "    return dsl.ContainerOp(\n",
+    "      name=\"minist_training_container\",\n",
+    "      image='gcr.io/{}/minist_training_kf_pipeline:latest'.format(PROJECT_ID),\n",
+    "      command=['python', '/app/app.py'],\n",
+    "      file_outputs={'outputs': '/output.txt'},\n",
+    "      arguments=['--bucket', bucket, '--model_file', model_file]\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create your workflow as a Python function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define your pipeline as a Python function. ` @kfp.dsl.pipeline` is a required decoration including `name` and `description` properties. Then compile the pipeline function. After the compilation is completed, a pipeline file is created."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define the pipeline\n",
+    "@dsl.pipeline(\n",
+    "   name='Minist pipeline',\n",
+    "   description='A toy pipeline that performs minist model training.'\n",
+    ")\n",
+    "def minist_container_pipeline(\n",
+    "    model_file: str = 'mnist_model.h5', \n",
+    "    bucket: str = GCS_BUCKET\n",
+    "):\n",
+    "    minist_train_op(model_file=model_file, bucket=bucket).apply(gcp.use_gcp_secret('user-gcp-sa'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "Experiment link <a href=\"https://kubeflow1.endpoints.kubeflow-pipeline-fantasy.cloud.goog/pipeline/#/experiments/details/a1bdd769-68c9-42eb-a48b-f239f2f0a94e\" target=\"_blank\" >here</a>"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "text/html": [
+       "Run link <a href=\"https://kubeflow1.endpoints.kubeflow-pipeline-fantasy.cloud.goog/pipeline/#/runs/details/f9ac8170-11a2-45ee-89de-c86490cce2f1\" target=\"_blank\" >here</a>"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "pipeline_func = minist_container_pipeline\n",
+    "pipeline_filename = pipeline_func.__name__ + '.pipeline.zip'\n",
+    "\n",
+    "compiler.Compiler().compile(pipeline_func, pipeline_filename)\n",
+    "#Submit a pipeline run\n",
+    "arguments = {\"model_file\":\"mnist_model.h5\",\n",
+    "             \"bucket\":GCS_BUCKET}\n",
+    "run_name = pipeline_func.__name__ + ' run'\n",
+    "experiment = client.create_experiment('python-functions-minist')\n",
+    "\n",
+    "run_result = client.run_pipeline(experiment.id, run_name, pipeline_filename, arguments)\n",
+    "\n",
+    "## Submit pipeline directly from pipeline function\n",
```

Comment:
Can you switch to using this by default. This is the modern API

Reply:
Thanks for the suggestion. I have changed all the notebooks to use `create_run_from_pipeline_func`